candies/cli/cli.py

```diff
@@ -0,0 +1,59 @@
+import sys
+from typing import List, Callable, Optional, Union
+
+from candies.cli.parsers import Parser, StandardParser
+
+
+class CLI:
+    """A command line interface.
+
+    Examples:
+        @cli
+        def main(a: Arg['description'],
+                 b: Arg['description'] = 'default'):
+            \"""Some help text.\"""
+            ...
+
+        if __name__ == '__main__':
+            main()
+    """
+
+    def __init__(self, main):
+        self._main = main
+
+    def __call__(self, args: List[str] = None, parser: Parser = None):
+        """Parses the specified arguments and calls the wrapped function."""
+
+        if parser is None:
+            parser = StandardParser(self._main)
+
+        args = parser.parse(args or sys.argv[1:])
+
+        # Note: this approach does not handle `--help`.
+        return self._main(**args)
+
+    def command(self, func):
+        raise NotImplementedError
+
+
+def cli(func: Optional[Callable] = None) -> Union[Callable, CLI]:
+    """Constructs a CLI from the specified function.
+
+    Args:
+        func (function, optional): A function to construct a CLI from.
+
+    Returns:
+        Either an instance of `CLI` if `func` was specified or a decorator
+        to wrap a function with.
+
+    Examples:
+        @cli
```

Comment:
Seems like this example only covers a scenario, when `cli` is acting like a decorator, but I'm not sure about the other case. Is it for arguments passing like `@cli(arg=value)`?

Reply:
Exactly! This may be used for configuring the CLI. Take a look at this example:
```python
@cli(prefix='-', epilogue='some epilogue')
def main(a: Arg):
    ...
```
(The parameters `prefix` and `epilogue` are taken from the standard `ArgumentParser`.)